tests/test_chain.py

```diff
@@ -303,3 +304,56 @@ def test_deposit_replacement(lianad, bitcoind):
     addr = lianad.rpc.getnewaddress()["address"]
     bitcoind.rpc.sendtoaddress(addr, 2)
     wait_for(lambda: len(lianad.rpc.listcoins()["coins"]) == 1)
+
+def test_rescan_and_recovery(lianad, bitcoind):
+    """Test user recovery flow"""
+    # Get initial_tip to use for rescan later
+    initial_tip = bitcoind.rpc.getblockheader(bitcoind.rpc.getbestblockhash())
+
+    # Start by getting a few coins
+    destinations = {
+        lianad.rpc.getnewaddress()["address"]: 0.5,
+    }
+    txid = bitcoind.rpc.sendmany("", destinations)
+    bitcoind.generate_block(1, wait_for_mempool=txid)
+    wait_for(
+        lambda: lianad.rpc.getinfo()["block_height"] == bitcoind.rpc.getblockcount()
+    )
+    assert len(lianad.rpc.listcoins()["coins"]) == 1
+
+    # Advance the blocktime by >2h in median-time past for rescan
+    added_time = 60 * 60 * 3
+    bitcoind.rpc.setmocktime(initial_tip["time"] + added_time)
+    bitcoind.generate_block(12)
+
+    # Clear lianad state
+    lianad.restart_fresh(bitcoind)
+    assert len(lianad.rpc.listcoins()["coins"]) == 0
```

Comment:
Yeah it's because when importing a descriptor on `bitcoind`, the wallet will always rescan the last two hours of blocks. This is what happens when we restart fresh: `lianad` will import the descriptor on bitcoind, it will in turn scan the past 2h of blocks and find the coins.

The "past two hours" is done using MTP. The difference between the time the block were mined and the MTP at the tip must be >2h. That's the reason for the snippet above:
```python
    # Advance the blocktime by >2h in median-time past for rescan
    added_time = 60 * 60 * 3
    bitcoind.rpc.setmocktime(initial_tip["time"] + added_time)
    bitcoind.generate_block(12)
```

(Which could by the way mine only 6 blocks instead of 12?)

Reply:
If I understand correctly, I don't need to call `lianad.rpc.startrescan()` because the coins have already been recovered after `restart_fresh` so I should be able to create a recovery transaction immediately.

Reply:
> (Which could by the way mine only 6 blocks instead of 12?)
If you are asking if I was able to mine just 6 blocks instead of 12 after setting the mock time, Yes. I was able to mine as little as 1 block.

Reply:
> If I understand correctly, I don't need to call lianad.rpc.startrescan() because the coins have already been recovered after restart_fresh so I should be able to create a recovery transaction immediately.

No, you need to test a situation as close as what the user would experience. The user would most likely recover after two days and therefore would need to rescan. This is why we mock this situation by mining blocks and advancing the time.